lib/woody/server/http.ex

```diff
@@ -65,38 +63,58 @@ defmodule Woody.Server.Http do
 
       @impl true
       def handle_function(function, args, ctx, {innermod, hdlopts}) do
-        {:ok, innermod.handle_function(function, args, ctx, hdlopts)}
-      end
-      def handle_function(function, args, ctx, innermod) do
-        handle_function(function, args, ctx, {innermod, nil})
+        try do
+          {:ok, innermod.handle_function(function, args, ctx, hdlopts)}
+        catch
+          :throw, %type{} = ex ->
+            {:exception, type, ex}
+        end
```

Comment:
```suggestion
        {:ok, innermod.handle_function(function, args, ctx, hdlopts)}
      rescue
        ex ->
          {:exception, ex.__struct__, ex}
      end
```

И можно переключиться на более eлексирный raise в хэндлерах 

```elixir
        raise WeaponFailure, code: "invalid_shift", reason: "Shifted into #{pos} position"
```

Reply:
Под это нужно генератор подпиливать. 🥲 Это не слишком сложно, просто спойлер, изменение будет нетривиальное.

Reply:
Возможно я что-то упускаю, но я не вижу мест, где надо подпилить генератор для этого  🍞

Reply:
У меня сейчас понимание такое: так как `raise` бизнесовых исключений станет неотличим от `raise` всех остальных эликсирных исключений, нужно генерировать код, который ловит _только_ бизнесовые исключения, и превращает их в reply.

Reply:
Логично. В качестве альтернативы можно вообще отказаться от использования исключений для бизнес ошибок в пользу `{:ok, _} | {:error, _}`. Мне кажется это лучше отражает то, что бизнес-ошибка это все таки штатная ситуация.